app/javascript/sound-engines/sampler-engine.js

```diff
@@ -1,18 +1,63 @@
 class SamplerEngine {
+  currentInstrumentIndex = -1;
+  instruments = [
+    {
+      name: 'Piano',
+      path: 'assets/sample.wav',
+    },
+  ];
+
   constructor() {
     this.name = 'Sampler';
-    this.instrument = '-';
+    this.changeInstrument();
+    this.audioCtx = new AudioContext();
   }
 
   onPadPressed(padIndex) {
     console.log(`Pad ${padIndex} pushed from the Sampler engine`);
 
-    // Implement here the logic to produce Sound based on the Pad Index (int)
+    // TODO: Play in correct notes
+    this._play(padIndex / 15);
   }
 
   changeInstrument() {
-    console.log('Changed instrument for the Sampler');
-    this.instrument = 'A new instrument!';
+    this.currentInstrumentIndex++;
+    if (this.currentInstrumentIndex > this.instruments.length - 1) {
+      this.currentInstrumentIndex = 0;
+    }
+    const instrumentData = this.instruments[this.currentInstrumentIndex];
+    this._loadSample(instrumentData.path).then(
+      (sample) => (this.sample = sample)
+    );
+    this.instrument = instrumentData.name;
+  }
+
+  _play(rate) {
+    if (this.sample == null) return;
+
+    const context = new AudioContext();
+    const source = context.createBufferSource();
+    const time = this.audioCtx.currentTime;
+    source.buffer = this.sample;
+    source.playbackRate.value = rate;
+    source.connect(context.destination);
```

Comment:
Does it work when using the previously intantiated audio context? 🤔 
```suggestion
    const source = this.audioCtx.createBufferSource();
    const time = this.audioCtx.currentTime;
    source.buffer = this.sample;
    source.playbackRate.value = rate;
    source.connect(this.audioCtx.destination);
```

Reply:
(this is supposed to improve performances a LOT as we do not ask the Web Browser to start a new audio interface for each note ~)

Reply:
@malparty my bad 😅

Reply:
@malparty updated at[ed784d6](https://github.com/malparty/nimbleat/pull/16/commits/ed784d6e1fa48a76c47f88883b01435d274369a6)
Included `drum` !!!